src/constants/analytics.ts

```diff
@@ -206,6 +208,15 @@ export type AnalyticsEventData<T extends AnalyticsEvent> =
         type: string;
         id: string;
       }
+    : T extends AnalyticsEvent.TransferNotification
+    ? {
+        type: TransferNotificationTypes | undefined;
+        toAmount: number | undefined;
+        timeSpent: Record<string, number> | number | undefined;
+        txHash: string;
+        status: string;
+        triggeredAt: number | undefined;
```

Comment:
nit: when a property can be `undefined`, can we write them as something like:
```
type?: TransferNotificationTypes;
toAmount?: number;
...
```

Reply:
hmm i thought that's intended if a property doesn't have to be passed in, right? In this case the property keys are always being passed in but their type values are optional